enhancements/agent-installer/simplified_cluster_operations.md

```diff
@@ -0,0 +1,425 @@
+---
+title: simplified-registry-less-cluster-operations
+authors:
+  - "@andfasano"
+reviewers:
+  - "@danielerez"
+  - "@oourfali"
+  - "@bfournie"
+  - "@rwsu"
+  - "@pawanpinjarkar"
+approvers:
+  - "@zaneb"
+api-approvers:
+  - TBD
+creation-date: 2025-07-07
+last-updated: 2025-07-07
+tracking-link: 
+  - "https://issues.redhat.com/browse/AGENT-1250"
+  - "https://issues.redhat.com/browse/VIRTSTRAT-60"
+see-also: []
+replaces: []
+superseded-by: []
+---
+
+# Simplified registry-less cluster operations for disconnected environments
+
+## Summary
+
+This enhancement proposal outlines an opinionated workflow to simplify
+on-premises cluster operations - such as installation, upgrade and node
+expansion - by eliminating the need to setup and configure an external
+registry mirror. This approach is specifically designed for disconnected
+environments and leverages existing tools such as Agent-based Installer,
+Assisted Installer and OpenShift Appliance.
+
+## Motivation
+
+One of the most recurring pain points when installing - and later managing -
+an OpenShift cluster in a disconnected scenario has been the requirement to
+setup an [additional registry](https://docs.redhat.com/en/documentation/openshift_container_platform/4.19/html/disconnected_environments/index)
+within the same environment. This task involves not only deploying and
+maintaining the registry service itself, but also mirroring the desired
+OpenShift release payload - and optionally some additional OLM operators.
+Such complexity made the initial deployment and further upgrades challenging
+and time-consuming, often resulting in a frustrating process for the less
+experienced users. Additionally, this approach was not considered suitable for
+resource-constrained environments - where the lack of dedicated hardware for
+hosting an external registry made it impractical.
+
+### User Stories
+
+* As an OpenShift administrator, I want to install a cluster into a
+  disconnected environment without setting up an external registry (and
+  optionally a selected set of OLM operators)
+* As an OpenShift administrator, I want to upgrade a cluster into a
+  disconnected environment without setting up an external registry (and
+  optionally a selected set of OLM operators)
+* As an OpenShift administrator, I want to add a node to an existing cluster in
+  a disconnected environment without setting up an external registry
+
+### Goals
+
+* Ensure successful installation/upgrade/adding new nodes in disconnected
+  environments with minimal requirements and no pre-existing registry
+* Offer a simplified installation experience for disconnected environments by
+  leveraging existing tools like the Agent-Based Installer, Assisted Installer
+  and OpenShift Appliance
+* Allow the user to optionally install/upgrade a predefined subset of OLM
```

Comment:
Since this is not currently covered in this enhancement (and will probably be a Whole Thing), maybe we should move this goal to a separate enhancement.

Reply:
Ok